middleware/keyauth/keyauth.go

```diff
@@ -34,17 +35,12 @@
 	cfg := configDefault(config...)
 
 	// Initialize
-	parts := strings.Split(cfg.KeyLookup, ":")
-	extractor := keyFromHeader(parts[1], cfg.AuthScheme)
-	switch parts[0] {
-	case query:
-		extractor = keyFromQuery(parts[1])
-	case form:
-		extractor = keyFromForm(parts[1])
-	case param:
-		extractor = keyFromParam(parts[1])
-	case cookie:
-		extractor = keyFromCookie(parts[1])
+	if cfg.CustomKeyLookup == nil {
+		var err error
+		cfg.CustomKeyLookup, err = SingleKeyLookup(cfg.KeyLookup, cfg.AuthScheme)
+		if err != nil {
+			panic(fmt.Errorf("unable to create lookup function: %w", err))
+		}
```

Comment:
@gaby I think that this suggestion should _be ignored_. It's not standard for the other middlewares to return (handler, error) tuples, **and** configDefault already panics in case of invalid config.

With the lint and test errors out of the way, I'm considering this PR ready to be squashed and merged unless you have any additional comments.

Reply:
@coderabbitai Not relevant, we want the program to exit